studio.py

```diff
@@ -14,14 +29,201 @@
+            ) as proxy_service:
+                app_service = APPS[self.app](proxy_service, self.certutil, self.binary)
+                app_service.start(site["url"])
+
+                if not site.get("login", None):
+                    for i in range(0, RECORD_TIMEOUT):
+                        print(i)
+                        time.sleep(1)
+                else:
+                    time.sleep(5)
+                    raw_input("Do user input and press <Return>")
+
+                app_service.screen_shot(site["screen_shot_path"])
+                self.information["app_info"] = app_service.app_information()
+                app_service.stop()
+
+            self.update_json_information(site)
+            self.generate_zip_file(site)
+            self.generate_manifest_file(site)
+
+    def parse_sites_json(self):
+        print("Parsing sites json")
+        sites = []
+        if self.sites is not None:
+            with open(self.sites, "r") as sites_file:
+                sites_json = json.loads(sites_file.read())
+                self.information["app"] = self.app.lower()
+
+            self.information["os_name"] = platform.system()
+            self.information["os_version"] = platform.release()
+
+            self.information["system"] = platform.system()
+            self.information["release"] = platform.release()
+            self.information["version"] = platform.version()
+            self.information["machine"] = platform.machine()
+            self.information["processor"] = platform.processor()
+
+            platform_name = platform.system().lower()
+
+            if isinstance(self.app, AbstractAndroidFirefox):
+
+                device = ADBAndroid()
+                self.information["ro.product.model"] = device.shell_output(
+                    "getprop ro.product.model"
+                )
+                self.information["ro.build.user"] = device.shell_output(
+                    "getprop ro.build.user"
+                )
+                self.information["ro.build.version.release"] = device.shell_output(
+                    "getprop ro.build.version.release"
+                )
+
+                platform_name = "".join(
+                    e for e in self.information["device_name"] if e.isalnum()
+                )
+
+            for site in sites_json:
+                label = site.get("label")
+                if label:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            site["domain"],
+                            label,
+                        ]
+                    )
+                else:
+                    name = "-".join(
+                        [self.proxy, platform_name, self.app.lower(), site["domain"]]
+                    )
+                site["path"] = os.path.join(self.path, name, name)
```

Comment:
Can you explain why we have `name` twice?

Reply:
we want to create a path like
```
/.../recording_name/recording_name.<extension>
```

having this will get files better organized after we recorded everything